src/db/nedb.js

```diff
@@ -63,6 +64,29 @@ class DBHelper {
     }
   }
 
+  /*
+  * Returns the fields of the object in one of the tables searched by the query.
+  * @param db The DB table.
+  * @param query {Object} The query by items.
+  * @param fields {Array} The fields to return for the found item in an array.
+  */
+  static async findOne(db, query, fields) {
+    let fieldsObj;
+    if (!_.isEmpty(fields)) {
+      fieldsObj = {};
+      _.each(fields, (field) => {
+        fieldsObj[field] = 1;
+      });
+    }
+
+    const found = await db.findOne(query, fieldsObj);
+    if (!found) {
+      const filename = db.nedb.filename;
```

Comment:
does destructuring not work?
`const { filename } = db.nedb;`

Reply:
fixed